lib/dues_calculator.rb

```diff
@@ -20,16 +20,19 @@ def self.dues_for_n_competitors(country_iso2, base_entry_fee_lowest_denomination
 
   def self.dues_per_competitor_in_usd(country_iso2, base_entry_fee_lowest_denomination, currency_code)
     country_band = CountryBand.find_by(iso2: country_iso2)&.number
+    country_band_detail = CountryBandDetail.find_by(number: country_band)
```

Comment:
You're completely ignoring your new `belongs_to` association here.

Reply:
Oh sorry, changed it.